### DISCUSSION TITLE
Popup module recommendations?

### DISCUSSION THREAD

**Discussion by bobhuf** [timber/timber#3075] [General]
I have an upcoming e-commerce project that requires a popup module to present offers to visitors. I'm curious if anyone has experience with developer-friendly solutions for this. What’s your go-to plugin, or have you opted to create a custom solution?

Thanks.

**Comment by rmens** [comment]
GetSiteControl is really nice for this. It's an external service but I like it a lot.

**Reply by bobhuf** [reply]
Thanks, my intention was to stay in the WP ecosystem but I had a look anyway. It seems to be blocked by ad managers by default so it really isn't an option I'm afraid.

**Comment by mmorris8** [comment]
When I do popups I use the newly widely available popover api.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/popover

**Comment by bobhuf** [comment]
Yes, I've been using the Popover API for any modal type of thing like mobile menu, search form, shopping cart etc but it's not so handy when having to set triggers or rules for marketing popups.

**Comment by francesco-bizen** [comment]
I generally setup my own popup pipeline within the theme. wheter it's a component in a page or a "popup" custom post type.
The engine to actually make the popup vary. Used to be bootstrap's modals, then alpine.js now i just like to leverage the default popover API with a sprinkle of JS.

Just steer clear from popup maker and other bloated plugins. Just build your own with the tech you prefer so you have total control over its functions. You can use ACF and some clever thinking to make them show on all the combinations of conditions you prefer.
